resource/geo-density-chart.js

```diff
@@ -0,0 +1,379 @@
+/**
+
+ GeoDensityChart:
+
+ Provide a logical layer between Leaflet and the Dataiku custom chart environment.
+
+ USAGE:
+
+     Initialise the map:
+         chartHandler = new GeoDensityChart();
+         chartHandler.initialiseMap();
+         chartHandler.addLeafletLayer();
+         chartHandler.addMousePosition();
+         chartHandler.addScatterPlotLayer();
+         chartHandler.addSearchTrigger();
+         chartHandler.addUpdateEvent();
+
+     Update parameters:
+         chartHandler.colorPalette = colorPalette;
+         chartHandler.intensity = configEvent.intensity;
+         chartHandler.radius = configEvent.radius;
+         chartHandler.gradient = convertPaletteToGradient(colorPalette);
+
+     Update visualisation:
+        chartHandler.render()
+
+ */
+
+function GeoDensityChart(){
+
+    let _initialised = false;
+    let _mapId = 'mapid';
+    let _mapPointer;
+    let _svgPointer;
+    let _quadtree;
+    let _closestMarker;
+    let _heatMapLayer;
+    let _position;
+    let _tooltip;
+    let _coreData = [];
+
+    let _intensity;
+    let _radius;
+    let _gradient;
+    let _colorPalette;
+
+    Object.defineProperty(this, 'emptyDisplay', {
+        get: function(){ return (!_coreData || _coreData.length === 0);}
+    });
+
+    Object.defineProperty(this, 'initialised', {
+        get: function(){ return _initialised;},
+        set: function(val){ _initialised = val;}
+    });
+
+    Object.defineProperty(this, 'coreData', {
+        get: function(){ return _coreData;},
+        set: function(val){ _coreData = val;}
+    });
+
+    Object.defineProperty(this, 'tooltip', {
+        get: function(){ return _tooltip;},
+        set: function(val){ _tooltip = val;}
+    });
+
+    Object.defineProperty(this, 'intensity', {
+        get: function(){ return _intensity;},
+        set: function(val){ _intensity = val;}
+    });
+
+    Object.defineProperty(this, 'radius', {
+        get: function(){ return _radius;},
+        set: function(val){ _radius = val;}
+    });
+
+    Object.defineProperty(this, 'gradient', {
+        get: function(){ return _gradient;},
+        set: function(val){ _gradient = val;}
+    });
+
+    Object.defineProperty(this, 'colorPalette', {
+        get: function(){ return _colorPalette;},
+        set: function(val){ _colorPalette = val;}
+    });
+
+    Object.defineProperty(this, 'mapPointer', {
+        get: function(){ return _mapPointer;}
+    });
+
+    /**
+     * Part of the initialisation of the chart
+     */
+    this.initialiseMap = function(){
+        _mapPointer = L.map(_mapId).setView([42, -17], 1);
```

Comment:
Use a constant to define default map coords
```suggestion
        const defaultCoords = [42, -17];
        _mapPointer = L.map(_mapId).setView(defaultCoords, 1);
```

Reply:
Done in3e4b842